### DISCUSSION TITLE
AXParameter example for ACAP native SDK 4

### DISCUSSION THREAD

**Discussion by eliangbb** [AxisCommunications/acap-native-sdk-examples#69] [General] [answered]
Hi, 
Does anybody have a sample application (whole folder) about using AXParameter? I am working on Native SDK 4.0 not 3.x. (BTW, There is an example for SDK3.x at https://www.axis.com/techsup/developer_doc/acap3/3.1/api/axparameter/html/axparameterexample_8c-example.html )

I tried to include axparameter.h in my project, but I got "No such file or directory" error for "#include <axsdk/axparameter.h>" or "#include <axparameter.h>". So, does SDK 4.0 still work for AXParameter? how to include the necessary header(s)? Again, where can I find newer sample for AXParameter for SDK4? 

Thx.

Liang

**Comment by pataxis** [comment, marked as answer]
Hi @eliangbb , many of the ax* libraries like `axhttp` and `axparameter` are not included in ACAP Native SDK (ACAP 4), see
https://axiscommunications.github.io/acap-documentation/docs/introduction/acap4-vs-acap3.html for more information.

If you need to use these libraries you should stay on ACAP 3 and you can find the ACAP 3 SDK documentation on https://help.axis.com/acap-3-developer-guide.

**Reply by eliangbb** [reply]
Hi @pataxis , thank you so much. 

I just tried using SDK 3.2 to build sample code in axparameterexample.c. But got error: "undefined reference to 'ax_parameter_new'" although I did include line "#include <axsdk/axparameter.h>".
(https://www.axis.com/techsup/developer_doc/acap3/3.1/api/axparameter/html/axparameterexample_8c-example.html)
So, anything wrong to show this error? Do you have the complete folder/files for this sample?

Another question:
I read the sdk guide you mentioned at https://help.axis.com/acap-3-developer-guide,
It mentions that "Parameters specific to the application can be pre-configured when the application is created or be added in runtime.", do you have more resource to show how pre-configure works? Does it mean that user could input info manually? I just created a new discussion about "user interacts with APP", kind of related to this question.

Thank you again.

**Reply by pataxis** [reply]
You do need to also add `axparameter` to your `Makefile`, something along the lines of
```Makefile
PKGS = glib-2.0 gio-2.0 axparameter
CFLAGS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --cflags $(PKGS))
LDLIBS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs $(PKGS))
```
where I followed the structure of `vdostream` example [Makefile](https://github.com/AxisCommunications/acap3-examples/blob/main/vdostream/app/Makefile). The `pkg-config` program will get the paths to the header files expanded for you during the compilation.

Did this answer your question?

The second question:
No, there is not really more resources than the API documentation. Also, in the **Note** just above that sentence it says

> Data that’s only used and altered by the application should not be handled by AxParameter. Such data can instead be handled by, for example, GLib GKeyFile.

We would recommend that you find another solution for handling application specific parameters. For manual input, there should also be many ways to handle this, the [web-server](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server) example shows one way and this example should also be compatible with ACAP 3 SDK.

**Comment by pataxis** [comment]
@eliangbb  A follow-up, we have recently launched axparameter as a supported API in ACAP version [4.13 release](https://axiscommunications.github.io/acap-documentation/docs/release-notes/4.13.html) where ACAP Native SDK has version 1.13.
